### DISCUSSION TITLE
This program should **NOT** auto update by default...

### DISCUSSION THREAD

**Discussion by ghost** [Devolutions/UniGetUI#2579] [General]
Why does this program think it gets special privledge to automatically update itself at startup when the purpose of the program is to manually update every other program, it shouldn't be managing it's own updates, updating should be done by winget *manually* like every other program. I can't trust a program that updates, changes, and breaks every other time I boot into Windows.

Automatic updates need to be opt in not opt out.

**Comment by ghost** [comment]
This program breaks so often from updates, causing me to have to delete the entire program and reinstall it. It should not update unless explicitly told to.

**Reply by ghost** [reply]
And I don't want to hear "you can turn it off in the settings". My point is that the setting needs to be disabled by default, no exceptions. Automatic updates are the exact thing I use this program to avoid.

**Comment by marticliment** [comment]
Most software autoupdates automatically. This is nothing new.

**Reply by Lockszmith-GH** [reply]
If I recall correctly, on the first launch of WinGetUI (way back when), there were a set of questions, is this not the case with UniGetUI now?

Having this in the initial wizard/setup process would probably be the best solution here - allowing you to keep the value as opt-out by default, but allowing it to be disabled on the first attempt.

It feels like OP is really passionate about this, as they posted about it, not once, but twich more after the post 🤭. (Just an FYI, it's very easy to edit a post/comment using the `...` menu on the top-right of the text block)

**Reply by marticliment** [reply]
Yes, the welcome wizard is planned, but I haven't ported it from Python to C# yet

**Reply by Lockszmith-GH** [reply]
Awesome.
@SheaFoxx, does that answer your concern?

**Reply by Lockszmith-GH** [reply]
However, I do think SheaFoxx has a point to an extent. If UniGetUI is installed with winget, it should have autoupdates off. If it's installed interactively, the choice should be presented.

That's the standard with package managers. (Scoop and Chocolaty adheres to this as well).
